File: app/src/main/java/Controller/ProductPicturesRecyclerViewAdapter.java
package Controller;

import android.content.Context;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.elsafa.R;
import com.google.firebase.firestore.Blob;

import java.util.List;

public class ProductPicturesRecyclerViewAdapter extends RecyclerView.Adapter<ProductPicturesRecyclerViewAdapter.ViewHolder> {
    private List<Blob> pics;
    private Context context;

    public ProductPicturesRecyclerViewAdapter(List<Blob> pics, Context context) {
        this.pics = pics;
        this.context = context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.offer_card_view, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.imageView.setImageBitmap(BitmapFactory.decodeByteArray(pics.get(position)
                .toBytes(), 0, pics.get(position).toBytes().length));
    }

    @Override
    public int getItemCount() {
        return pics.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imageView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.item_image);
        }
    }
}

File: app/src/main/java/com/whiteside/cafe/adapter/ProductPicturesRecyclerViewAdapter.kt
package com.whiteside.cafe.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.firestore.Blob
import com.whiteside.cafe.R
import com.whiteside.cafe.utils.BlobBitmap

class ProductPicturesRecyclerViewAdapter(private val pics: MutableList<Blob>) :
    RecyclerView.Adapter<ProductPicturesRecyclerViewAdapter.ViewHolder?>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.offer_card_view, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.imageView.setImageBitmap(BlobBitmap.convertBlobToBitmap(pics[position]))
    }

    override fun getItemCount(): Int {
        return pics.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imageView: ImageView = itemView.findViewById(R.id.item_image)
    }
}